integrationTests/IntegrationTest.php

```diff
@@ -4,17 +4,51 @@
 
 use CreateDonationTest;
 use IntegrationTests;
+use LosMiddleware\RateLimit\RateLimitMiddleware;
+use MatchBot\Application\Auth\DonationRecaptchaMiddleware;
 use PHPUnit\Framework\TestCase;
 use Prophecy\Prophecy\ObjectProphecy;
 use Psr\Container\ContainerInterface;
+use Psr\Http\Message\ResponseInterface;
+use Psr\Http\Message\ServerRequestInterface;
+use Psr\Http\Server\MiddlewareInterface;
+use Psr\Http\Server\RequestHandlerInterface;
+
 use Slim\App;
+use Slim\Factory\AppFactory;
 use Stripe\StripeClient;
 
 abstract class IntegrationTest extends TestCase
 {
     public static ?ContainerInterface $integrationTestContainer = null;
     public static ?App $app = null;
 
+    public function setUp(): void
+    {
+        parent::setUp();
+
+        $noOpMiddlware = new class implements MiddlewareInterface {
+            public function process(ServerRequestInterface $request, RequestHandlerInterface $handler): ResponseInterface
+            {
+                return $handler->handle($request);
+            }
+        };
+
+        $container = require __DIR__ . '/../bootstrap.php';
+        IntegrationTest::setContainer($container);
+        $container->set(DonationRecaptchaMiddleware::class, $noOpMiddlware);
+        $container->set(RateLimitMiddleware::class, $noOpMiddlware);
+        $container->set(\Psr\Log\LoggerInterface::class, new \Psr\Log\NullLogger());
+
+        AppFactory::setContainer($container);
+        $app = AppFactory::create();
+
+        $routes = require __DIR__ . '/../app/routes.php';
+        $routes($app);
+
+        self::setApp($app);
```

Comment:
moved this from the bootstrap to the setup because it seemed like some stuff was persisting between tests causing a duplicate db key error